include/pybind11/cast.h

```diff
@@ -960,9 +960,15 @@ template <typename type> class type_caster<std::reference_wrapper<type>> {
 private:
     using caster_t = make_caster<type>;
     caster_t subcaster;
-    using subcaster_cast_op_type = typename caster_t::template cast_op_type<type>;
-    static_assert(std::is_same<typename std::remove_const<type>::type &, subcaster_cast_op_type>::value,
-            "std::reference_wrapper<T> caster requires T to have a caster with an `T &` operator");
+    using reference_t = typename std::add_lvalue_reference<type>::type;
+    using subcaster_cast_op_type =
```

Comment:
If this is changed, why does a few lines later

```c++
operator std::reference_wrapper<type>() { return subcaster.operator subcaster_cast_op_type&(); }
```

not become

```c++
operator std::reference_wrapper<type>() { return subcaster.operator subcaster_cast_op_type(); }
```
?

Reply:
Done. Indeed I missed that and it is no longer required.  From a C++ perspective they are equivalent, though.